src/Module.zig

```diff
@@ -3735,7 +4108,17 @@ fn semaDecl(mod: *Module, decl_index: Decl.Index) !bool {
     };
     decl.has_tv = true;
     decl.analysis = .complete;
-    decl.generation = mod.generation;
+
+    const result: SemaDeclResult = if (old_has_tv) .{
+        .invalidate_decl_val = !decl.ty.eql(old_ty, mod) or !decl.val.eql(old_val, decl.ty, mod),
+        .invalidate_decl_ref = !decl.ty.eql(old_ty, mod) or
```

Comment:
Thank goodness these are just integer equality checks.

Reply:
Is there any point in really having `Type.eql` anymore? At least `Value.eql` has a useful assertion, but `Type.eql` just seems plainly unnecessary.

Reply:
No, it can be removed. The only reason it wasn't changed already was to reduce churn / limit conflicts.